### ISSUE TITLE
SyntaxError at newline \n

### ISSUE THREAD

**Issue by kenn** [beenotung/best-effort-json-parser#4]
parse throws a SyntaxError at a newline character `\n`.

```js
repl> data = parse('{"a":"\n"}')
Uncaught SyntaxError: Bad control character in string literal in JSON at position 1
repl> data = parse('{"a":"\\n"}')
{ a: '\n' }
```

edit: never mind, it's actually double-escaping matter!

**Comment by beenotung** [comment]
Nice catch, maybe this library can do better effort to handle this kind of uneacaped input as well

**Comment by beenotung** [comment]
The update is released to npm as `best-effort-json-parser@1.1.0` :rocket: 

Feel free to re-open the issue if it doesn't address your case

**Comment by kenn** [comment]
It's fixed indeed, thank you!

**Comment by kenn** [comment]
Turns out, I'm still getting the errors when deployed to the app.

This is what I got from GPT-4 in JSON mode (edited for clarity and anonymity):

```json
{
  "essay": "From the bustling corridors of ~, my journey in healthcare has been driven by ~, but also my resolve to make a tangible difference in this critical area of global health.\n\nDuring my tenure ..."
}
```

The error occurred right after `global health` and before printing `.\n\n`

I tried this with v1.1.0:

```js
node> parse('{"a":"\')
parse('{"a":"\')
      ^^^^^^^^^^
Uncaught SyntaxError: Invalid or unexpected token

node> parse('{"a":"\n')
Uncaught SyntaxError: Bad control character in string literal in JSON at position 1

node> parse('{"a":"\n"')
{ a: '\n' }
```

Note that LLMs give you a stream of text token-by-token, so there may be a backslash at the end of a string, as well as a control character before the closing quote.

**Comment by MartinCura** [comment]
Hi, I'm getting a similar issue trying to think how to fix it. The problem is that we can't have `\n` (a plain control character for a newline) inside a JSON string. If we replace all `\n`s with `\\n` then we'll also replace the ones that are just "prettifying" the JSON, which should stay as they are.

I was having the issue with 1.0.1 and after upgrading to 1.1.0 now i don't get the error if i replace the `\n`s but then the `\\n`s are considered part of the keys, so i may get a key such as `\\n  "content"` which is clearly not what i want. I could monkey-patch this but 🤔 

Will also try prompting better but would love for this library to be able to handle these cases. Can lend a hand if it helps!

**Comment by MartinCura** [comment]
Created a fix to my issues here: https://github.com/beenotung/best-effort-json-parser/pull/5

@kenn does this do it for you too?

**Comment by beenotung** [comment]
The pull request from @MartinCura did address the examples given by @kenn.
I've merged the PR and released to npm as v1.1.1 :rocket: 
Thank you for your helps :heart:

**Comment by MartinCura** [comment]
Excellent, thank you for the quick review!
